### PR TITLE
Issue/3334 fix skipping of posts when doing infinite scroll

### LINKED ISSUE #3334: Infinite scroll through posts skips dates on second refresh
Reported in helpshift chat #14510 

Issue is on iOS with App version 4.8

Tested and issue is not present on App version 4.7.1

Steps to recreate:
1) In a WordPress.com hosted site, go to Blog Posts with lots of posts over a wide date range
2) Scroll for the refresh of the Posts List.
3) See date jump.

This does not appear to be affecting WordPress.org sites.

### PR DESCRIPTION
**Pull request by SergioEstevao**
Closes #3334 

cc @aerych and @koke , can you please review

How to test:

 * Open a blog posts and scrolls down the list. You shouldn't see post dates being skipped.

This was only happening to user that have multiple blogs with very different post dates.

### CODE REVIEW

**Comment by aerych** [comment]
Nice :)  
:shipit:

File: WordPress/Classes/Services/PostService.m [hunk 1]
```diff
@@ -145,12 +145,13 @@ - (void)loadMorePostsOfType:(NSString *)postType
...
         NSString *entityName = [postType isEqualToString:PostServiceTypePage] ? NSStringFromClass([Page class]) : NSStringFromClass([Post class]);
         NSFetchRequest *request = [NSFetchRequest fetchRequestWithEntityName:entityName];
-        request.predicate = [NSPredicate predicateWithFormat:@"date_created_gmt != NULL"];
+        request.predicate = [NSPredicate predicateWithFormat:@"date_created_gmt != NULL AND blog=%@", blog];
         NSSortDescriptor *sortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"date_created_gmt" ascending:YES];
         request.sortDescriptors = @[sortDescriptor];
         Post *oldestPost = [[self.managedObjectContext executeFetchRequest:request error:nil] firstObject];
         if (oldestPost.date_created_gmt) {
             options[@"before"] = [oldestPost.date_created_gmt WordPressComJSONString];
+            options[@"order"] = @"desc";
```
Comment (koke) [inline WordPress/Classes/Services/PostService.m]: If we're setting `order`, I'd also explicitly set `order_by`, just  to be safe (and more readable)

**Comment by SergioEstevao** [comment]
@koke ready for another look.

File: WordPress/Classes/Services/PostService.m [hunk 2]
```diff
@@ -135,6 +135,16 @@ - (void)syncPostsOfType:(NSString *)postType
                    }];
 }
 
+- (Post *)oldesPostOfType:(NSString *)postType forBlog:(Blog *)blog {
```
Comment (koke) [inline WordPress/Classes/Services/PostService.m]: typo: "oldes" -> "oldest"

**Comment by koke** [comment]
Other than the typo in the method name, :shipit:

**Pull request merged**